src/HDWallet.cpp

```diff
@@ -146,22 +147,46 @@ std::optional<std::string> HDWallet::getAddressFromExtended(const std::string& e
 
 namespace {
     HDNode getNode(const HDWallet& wallet, uint32_t purpose, uint32_t coin) {
-        auto node = getMasterNode(wallet);
+        HDNode node;
+        switch (coin) {
+            case TWCoinTypeAion: node = getMasterNodeWithED25519(wallet); break;
+            default: node = getMasterNode(wallet);
+        }
+        
         hdnode_private_ckd(&node, purpose | 0x80000000);
         hdnode_private_ckd(&node, coin | 0x80000000);
         return node;
     }
 
     HDNode getNode(const HDWallet& wallet, uint32_t purpose, uint32_t coin, uint32_t account, uint32_t change, uint32_t address) {
-        auto node = getMasterNode(wallet);
+        HDNode node;
+        switch (coin) {
+            case TWCoinTypeAion: node = getMasterNodeWithED25519(wallet); break;
+            default: node = getMasterNode(wallet);
+        }
+        
         hdnode_private_ckd(&node, purpose | 0x80000000);
         hdnode_private_ckd(&node, coin | 0x80000000);
         hdnode_private_ckd(&node, account | 0x80000000);
-        hdnode_private_ckd(&node, change);
-        hdnode_private_ckd(&node, address);
+        
+        auto correctedChange = change;
+        auto correctedAddress = address;
+        if (coin == TWCoinTypeAion) {
```

Comment:
do `switch` here as well?

Reply:
I would find a way to not have a switch for coins in this function, @alejandro-isaza might have more insight on this

Reply:
we may not need change / address for curve ed25519 this? https://github.com/satoshilabs/slips/blob/master/slip-0010.md

Reply:
I did `switch` for now.